api/src/main/proto/stellarstation/api/v1/groundstation/plan-execution.proto

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2019 Infostellar, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+syntax = "proto3";
+
+import "google/protobuf/timestamp.proto";
+import "stellarstation/api/v1/stellarstation.proto";
+import "stellarstation/api/v1/transport.proto";
+
+package stellarstation.api.v1.groundstation;
+
+option go_package = "github.com/infostellarinc/go-stellarstation/api/v1/groundstation";
+
+option java_multiple_files = true;
+option java_outer_classname = "PlanExecution";
+option java_package = "com.stellarstation.api.v1.groundstation";
+
+// A life cycle event of a plan.
+message PlanLifeCycleEvent {
+  oneof Event {
+    // The plan has begun. Sent soon after the plan's `start_time'. The client should start setting
+    // up the ground station facility for the plan and be ready by `aos_time'.
+    PlanWillBegin begin = 1;
+
+    // The plan has ended. Sent when the plan's final status is determined. The client should tear
+    // down the ground statoin facility by the plan's `end_time'.
+    PlanDidEnd end = 2;
+  }
+}
+
+// An event notifying a plan has begun.
+message PlanWillBegin {
+  // Predicted coordinates of the tracked satellite for every second between AOS and LOS. This
```

Comment:
Any reason for 1 second? Does this mean that if a client needs more granularity (anything beyond UHF, I think?), they should ignore this field and get the TLE from the plan ID?

Reply:
Thanks for the feedback. I chose 1 second only because it matches the current implementation. What is the desired granularity for higher bands? I can make the granularity configurable but it incurs additional implementation cost. Can we have the client do interpolation on their own?

Reply:
This is beyond my knowledge. @vaelen @CanoRoger